Lib/glyphsLib/__main__.py

```diff
@@ -14,48 +14,185 @@
 
-import sys
 import argparse
+import os
+import sys
 
 import glyphsLib
 
 
-description = """\n
-Converts a Glyphs.app source file into UFO masters
-or UFO instances and MutatorMath designspace.
-"""
-
-
-def parse_options(args):
-    parser = argparse.ArgumentParser(description=description)
-    parser.add_argument("--version", action="version",
-                        version='glyphsLib %s' % (glyphsLib.__version__))
-    parser.add_argument("-g", "--glyphs", metavar="GLYPHS", required=True,
-                        help="Glyphs file to convert.")
-    parser.add_argument("-m", "--masters", metavar="MASTERS",
-                        default="master_ufo",
-                        help="Ouput masters UFO to folder MASTERS. "
-                             "(default: %(default)s)")
-    parser.add_argument("-n", "--instances", metavar="INSTANCES", nargs="?",
-                        const="instance_ufo", default=None,
-                        help="Output and generate interpolated instances UFO "
-                             "to folder INSTANCES. "
-                             "(default: %(const)s)")
-    parser.add_argument("-r", "--round-instances", action="store_true",
-                        help="Apply integer rounding to all geometry when "
-                             "interpolating")
+def glyphs2ufo(args=None):
+    """Converts a Glyphs.app source file into UFO masters or UFO instances and
+    MutatorMath designspace."""
+    parser = argparse.ArgumentParser(description=glyphs2ufo.__doc__)
+    parser.add_argument(
+        "--version",
+        action="version",
+        version="glyphsLib %s" % (glyphsLib.__version__))
+    parser.add_argument(
+        "glyphs_file",
+        metavar="GLYPHS_FILE",
+        help="Glyphs file to convert.",
+    )
+    parser.add_argument(
+        "-m",
+        "--output-dir",
+        default=None,
+        help=("Output directory of masters and designspace file. "
+              "(default: directory of Glyphs file)"),
+    )
+    parser.add_argument(
+        "-n",
+        "--instance-dir",
+        default=None,
+        help=(
+            "Output directory of instances. (default: output_dir/instance_ufos"
+            "). This sets the file path for instances inside the designspace "
+            "file."),
+    )
+
+    group = parser.add_argument_group('Roundtripping between Glyphs and UFOs')
+    group.add_argument(
+        "--no-preserve-glyphsapp-metadata",
+        action="store_false",
+        help=("Skip preserving Glyphs metadata in master UFOs and designspace "
+              "file, which would be used to minimize differences when "
+              "roundtripping between Glyphs and UFOs."),
+    )
+    group.add_argument(
+        "--propagate-anchors",
+        action="store_true",
+        help=("Copy anchors from underlying components to actual "
+              "glyph. Glyphs would do this implicitly, only use if you need "
+              "100% control over all anchors."),
```

Comment:
The `%` needs to be escaped here, otherwise running `glyph2ufo -h` fails for me with:
```
Traceback (most recent call last):
  File "/home/khaled/.local/bin/glyphs2ufo", line 11, in <module>
    sys.exit(glyphs2ufo())
  File "/home/khaled/.local/lib/python3.6/site-packages/glyphsLib/__main__.py", line 75, in glyphs2ufo
    options = parser.parse_args(args)
  File "/usr/lib/python3.6/argparse.py", line 1730, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/usr/lib/python3.6/argparse.py", line 1762, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib/python3.6/argparse.py", line 1968, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/usr/lib/python3.6/argparse.py", line 1908, in consume_optional
    take_action(action, args, option_string)
  File "/usr/lib/python3.6/argparse.py", line 1836, in take_action
    action(self, namespace, argument_values, option_string)
  File "/usr/lib/python3.6/argparse.py", line 1020, in __call__
    parser.print_help()
  File "/usr/lib/python3.6/argparse.py", line 2362, in print_help
    self._print_message(self.format_help(), file)
  File "/usr/lib/python3.6/argparse.py", line 2346, in format_help
    return formatter.format_help()
  File "/usr/lib/python3.6/argparse.py", line 282, in format_help
    help = self._root_section.format_help()
  File "/usr/lib/python3.6/argparse.py", line 213, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "/usr/lib/python3.6/argparse.py", line 213, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "/usr/lib/python3.6/argparse.py", line 213, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "/usr/lib/python3.6/argparse.py", line 213, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "/usr/lib/python3.6/argparse.py", line 519, in _format_action
    help_text = self._expand_help(action)
  File "/usr/lib/python3.6/argparse.py", line 606, in _expand_help
    return self._get_help_string(action) % params
TypeError: %c requires int or char
```

Reply:
to escape, add another `%` sign after `100%`, as in `100%%`

Reply:
I replaced `100%` by `full` in my buffer.